backend/src/main/java/de/jonashackt/springbootvuejs/controller/BackendController.java

```diff
@@ -52,12 +52,4 @@ public User getUserById(@PathVariable("id") long id) {
         return SECURED_TEXT;
     }
 
-    // Forwards all routes to FrontEnd except: '/', '/index.html', '/api', '/api/**'
-    // Required because of 'mode: history' usage in frontend routing, see README for further details
-    @RequestMapping(value = "{_:^(?!index\\.html|api).*$}")
```

Comment:
The `forward` in this rule did not work for two reasons

1. it was placed int he wrong `RequestMapping` context as it would look for the pattern only after `/api`
2. it was placed in a `RestController` instead of a `Controller` that does not support returning a `forwad:/`

Although the filter probably does work the easier solution would be to just fix 1) & 2)

Reply:
Thanks for the info, I didn't knew about it! I'll see if it works and I'll update the PR in the mean time

Reply:
@hemeroc I've tried to follow your idea, you were right about the using Controller instead of RestController but unfortunately, this solution only works on one level URLs like this `/login` but not for this `/auth/login` or any URL that has many deep level path as you can see the tests here: https://github.com/jonashackt/spring-boot-vuejs/blob/86785f7ce6c7e77737a3e930d9c1e8897f89d63c/backend/src/test/java/de/jonashackt/springbootvuejs/SpaRedirectTest.java#L34-L41

Do you have any idea on how to match URLs with multiple levels?

Reply:
I've found a way to accomplish using `@Controller` instead of `Filter` I'll update as soon as possible

Reply:
> I've found a way to accomplish using `@Controller` instead of `Filter` I'll update as soon as possible

when you will update :)

Reply:
@isneezy great contribution 👍 Sorry, didn't find the time earlier.